src/Auth.js

```diff
@@ -128,22 +128,33 @@ Auth.prototype._loadRoles = function() {
     };
     // First get the role ids this user is directly a member of
     var query = new RestQuery(this.config, master(this.config), '_Role', restWhere, {});
-    return query.execute().then((response) => {
+    const p = Parse.Promise.when(
+      new Parse.Query(Parse.Role)
```

Comment:
@flovilmart if you're ok with the general direction of the change, I'd like to discuss this issue with you....